app/models/manageiq/providers/kubernetes/container_manager/metrics_capture/capture_context.rb

```diff
@@ -0,0 +1,190 @@
+      when ContainerGroup then collect_group_metrics
+      else raise TargetValidationError, "Validation error: unknown target"
+      end
+    end
+
+    private
+
+    CPU_NANOSECONDS = 1e09
+
+    def target_name
+      "#{@target.class.name.demodulize}(#{@target.id})"
+    end
+
+    def validate_target
+      unless @ext_management_system
+        raise TargetValidationError, "Validation error: ems not defined"
+      end
+      unless @node_cores.to_i > 0
+        raise TargetValidationError, "Validation error: cores not defined"
+      end
+      unless @node_memory.to_i > 0
+        raise TargetValidationError, "Validation error: memory not defined"
+      end
+    end
+
+    def collect_node_metrics
+      cpu_resid = "machine/#{@target.name}/cpu/usage"
+      process_cpu_counters_rate(fetch_counters_rate(cpu_resid))
+
+      mem_resid = "machine/#{@target.name}/memory/usage"
+      process_mem_gauges_data(fetch_gauges_data(mem_resid))
+    end
+
+    def collect_container_metrics
+      group_id = @target.container_group.ems_ref
+
+      cpu_resid = "#{@target.name}/#{group_id}/cpu/usage"
+      process_cpu_counters_rate(fetch_counters_rate(cpu_resid))
+
+      mem_resid = "#{@target.name}/#{group_id}/memory/usage"
+      process_mem_gauges_data(fetch_gauges_data(mem_resid))
+    end
+
+    def collect_group_metrics
+      group_id = @target.ems_ref
+
+      cpu_counters = @target.containers.collect do |c|
+        fetch_counters_rate("#{c.name}/#{group_id}/cpu/usage")
+      end
+      process_cpu_counters_rate(compute_summation(cpu_counters))
+
+      mem_gauges = @target.containers.collect do |c|
+        fetch_gauges_data("#{c.name}/#{group_id}/memory/usage")
+      end
+      process_mem_gauges_data(compute_summation(mem_gauges))
+    end
+
+    def hawkular_client
+      @client ||= Hawkular::Metrics::Client.new(
+        hakwular_entrypoint, hakwular_credentials, hakwular_options)
+    end
+
+    def hakwular_entrypoint
+      worker_class = ManageIQ::Providers::Kubernetes:: \
+                     ContainerManager::MetricsCollectorWorker
+      URI::HTTPS.build(
+        :host => @ext_management_system.hostname,
+        :port => worker_class.worker_settings[:hawkular_port],
+        :path => worker_class.worker_settings[:hawkular_path])
+    end
+
+    def hakwular_credentials
+      {:token => @ext_management_system.try(:authentication_token)}
+    end
+
+    def hakwular_options
+      {:tenant     => @tenant,
+       :verify_ssl => @ext_management_system.verify_ssl_mode}
+    end
+
+    def fetch_counters_rate(resource)
+      compute_derivative(fetch_counters_data(resource))
+    end
+
+    def fetch_counters_data(resource)
+      sort_and_normalize(
+        hawkular_client.counters.get_data(
+          resource,
+          :starts         => (@start_time.to_i - @interval) * 1000,
+          :bucketDuration => "#{@interval}s"))
+    rescue SystemCallError => e
+      raise CollectionFailure, e.message
```

Comment:
@Fryguy on the contrary, at the moment the backtrace is more descriptive (the previous backtrace is not lost), e.g.

    class MyException < RuntimeError; end

    def test_exception
      raise RuntimeError, "Test Exception"
    rescue RuntimeError => e
      raise MyException, e.message
    end

    test_exception

In fact:

    $ ruby test_exception.rb 
    test_exception.rb:6:in `rescue in test_exception': Test Exception (MyException)
            from test_exception.rb:4:in `test_exception'
            from test_exception.rb:9:in `<main>'

If you're asking to change it to:

    raise CollectionFailure, e.message, e.backtrace

that would indeed hide the last entry of the backtrace.